src/language-yaml/printer-yaml.js

```diff
@@ -91,7 +91,12 @@ function genericPrint(path, options, print) {
 
   if (hasMiddleComments(node)) {
     parts.push([
-      node.middleComments.length === 1 ? "" : hardline,
+      !anchor ||
+      (node.middleComments.length === 1 &&
```

Comment:
Sorry for the wait. This condition looks too specific. What if there are multiple comments?

**Prettier pr-11395**
[Playground link](https://deploy-preview-11395--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEBrOBPAjEgBAMgBM4AzAQwFcAbGXAYl0gFsmEYsAdKXe3AFQAWASwDOudBlwA6GVx4iKAIwk5cANzJUKcTlC4SATEjm4APKbwAqYuWowTPXPswBmPA2atYBkx4gs2FxMFZVc8DS04EAAaEAgABxghaBFkUDIAJwyIAHcABUyEVJRNHLIMVNjFDLIwdBgAZXjaoSgAc2QYDO1YuCZFOEJiQgAZMnaKMja4ADEIDKYyGCT25BBKGAgYkAEYJioAdWF4EWawOAaioSS1a4w1sBFKkFaROAyYPJq2xeRyKjesQAViIAB4AIRqdTgjTIrBGrSiSH+gJAINBDVabSocAAihQIPA-ppUc0Mm8MmsMHCqNt4hlWjADkJCDABMgABwABli9IgbwONXia3pcApaiisQAjgT4F8EsV1iIALRQOCDQbbDJwGVCbVfKa-ZEkuCxN5MISdbqmkAiLE4-GEpEom0wMiKZms9lIAyxLpkIRULEAYX8RpAYoArNsKG8+O7ii7YmptABJKDEWANMAMxIAQQzDRgGBxxIBcAAvhWgA)
<!-- prettier-ignore -->
```sh
--parser yaml
```

**Input:**
<!-- prettier-ignore -->
```yaml
key1: &default # comment1
  # This key ...
  subkey1: value1

key2:
  <<: *default
     
key3: # comment2
  # comment3
  subkey3: value
```

**Output:**
<!-- prettier-ignore -->
```yaml
key1: &default
  # comment1
  # This key ...
  subkey1: value1

key2:
  <<: *default

key3: # comment2
  # comment3
  subkey3: value

```